packages/builtin/string.pony

```diff
@@ -971,27 +971,45 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
     that_offset: ISize = 0, ignore_case: Bool = false): Compare
   =>
     """
-    Starting at this + offset, compare n bytes with that + offset.
+    Lexically compare at most `n` bytes of the substring of `this` starting at
+    `offset` with the substring of `that` starting at `that_offset`. The
+    comparison is case sensitive unless `ignore_case` is `true`.
+
+    If the substring of `this` is a proper prefix of the substring of `that`,
+    then `this` is `Less` than `that`. Likewise, if `that` is a proper prefix of
+    `this`, then `this` is `Greater` than `that`.
+
+    Both `offset` and `that_offset` can be negative, in which case the offsets
+    are computed from the end of the string.
+
+    If `n + offset` is greater than the length of `this`, or `n + that_offset`
+    is greater than the length of `that`, then the number of positions compared
+    will be reduced to the length of the longest substring.
+
+    Needs to be made UTF-8 safe.
     """
-    var i = n
     var j: USize = offset_to_index(offset)
     var k: USize = that.offset_to_index(that_offset)
-
-    if (j + n) > _size then
-      return Less
-    elseif (k + n) > that._size then
-      return Greater
-    end
+    var i = n.min( (_size - j).max(that._size - k) )
```

Comment:
Can you remove the extra whitespace within the parenthesis of the `min` call here?  Just to match the code style of the rest of the codebase.